File: Find All Numbers Disappeared in an Array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        numset = set(nums);
        ans = []
        for i in range(1, len(nums) + 1):
            if i not in numset:
                ans.append(i)
                
                
        return ans
        

File: Find All Numbers Disappeared in an Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        int [] visited = new int [n];
        
        for (int i = 0; i < n; i ++){
            int cur = nums[i];
            
            visited[cur - 1] = 1;
        }
        
        List<Integer> ans = new ArrayList<Integer>();
        for (int i = 0; i < n; i ++){
            if (visited[i] == 0){
                
                ans.add(i + 1);
            }
            
        }
        
        
        return ans;
    }
}